src/app/AppLayout/AppLayout.tsx

```diff
@@ -173,18 +173,22 @@ const AppLayout: React.FunctionComponent<IAppLayout> = ({children}) => {
       () => setShowUserIcon(true)
     );
     return () => sub.unsubscribe();
-  }, [serviceContext, serviceContext.login]);
+  }, [serviceContext, serviceContext.login, setShowUserIcon]);
 
   React.useEffect(() => {
     const sub = serviceContext.login.loggedOut().subscribe(
       () => setShowUserIcon(false)
     );
     return () => sub.unsubscribe();
-  }, [serviceContext, serviceContext.login]);
+  }, [serviceContext, serviceContext.login, setShowUserIcon]);
 
-  const handleLogout = () => serviceContext.login.setLoggedOut();
+  const handleLogout = React.useCallback(() =>
+    serviceContext.login.setLoggedOut(),
+    [serviceContext.login]);
 
-  const handleUserInfoToggle = () => setShowUserInfoDropdown(!showUserInfoDropdown);
+  const handleUserInfoToggle = React.useCallback(() =>
+    setShowUserInfoDropdown(!showUserInfoDropdown),
```

Comment:
I think in this case, when toggling state using a React hook, it's better to do: `setShowUserInfoDropdown(v => !v)` - ie pass a mutation function to the state setter, rather than read the actual state "externally" from the setter and modify it from there.